pilot/pkg/security/authz/model/matcher/header_test.go

```diff
@@ -59,3 +60,78 @@ func TestHeaderMatcher(t *testing.T) {
 		}
 	}
 }
+
+func TestPathMatcher(t *testing.T) {
+	testCases := []struct {
+		Name   string
+		V      string
+		Expect *envoy_matcher.PathMatcher
+	}{
+		{
+			Name: "exact match",
+			V:    "/productpage",
+			Expect: &envoy_matcher.PathMatcher{
+				Rule: &envoy_matcher.PathMatcher_Path{
+					Path: &envoy_matcher.StringMatcher{
+						MatchPattern: &envoy_matcher.StringMatcher_Exact{
+							Exact: "/productpage",
+						},
+					},
+				},
+			},
+		},
+		{
+			Name: "prefix match",
+			V:    "/prefix*",
+			Expect: &envoy_matcher.PathMatcher{
+				Rule: &envoy_matcher.PathMatcher_Path{
+					Path: &envoy_matcher.StringMatcher{
+						MatchPattern: &envoy_matcher.StringMatcher_Prefix{
+							Prefix: "/prefix",
+						},
+					},
+				},
+			},
+		},
+		{
+			Name: "suffix match",
+			V:    "*suffix",
+			Expect: &envoy_matcher.PathMatcher{
+				Rule: &envoy_matcher.PathMatcher_Path{
+					Path: &envoy_matcher.StringMatcher{
+						MatchPattern: &envoy_matcher.StringMatcher_Suffix{
+							Suffix: "suffix",
+						},
+					},
+				},
+			},
+		},
+		{
+			Name: "wildcard match",
+			V:    "*",
+			Expect: &envoy_matcher.PathMatcher{
+				Rule: &envoy_matcher.PathMatcher_Path{
+					Path: &envoy_matcher.StringMatcher{
+						MatchPattern: &envoy_matcher.StringMatcher_SafeRegex{
+							SafeRegex: &envoy_matcher.RegexMatcher{
+								Regex: ".+",
+								EngineType: &envoy_matcher.RegexMatcher_GoogleRe2{
+									GoogleRe2: &envoy_matcher.RegexMatcher_GoogleRE2{},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			actual := PathMatcher(tc.V)
+			if !reflect.DeepEqual(*tc.Expect, *actual) {
+				t.Errorf("%s: expecting %v, but got %v", tc.Name, *tc.Expect, *actual)
```

Comment:
nit. Name is already shown in the test case, you can skip it in the error message.